File: src/main/java/org/khun/leetcode/BinaryTreeGeneral/InvertBinaryTree.java
package org.khun.leetcode.BinaryTreeGeneral;

public class InvertBinaryTree {

    public TreeNode invertTree(TreeNode root) {
        if (root == null) return  null;
        TreeNode temp = root.left;
        root.left = invertTree(root.right);
        root.right = invertTree(temp);
        return root;
    }
    public static void main(String[] args) {

    }
}

File: src/main/java/org/khun/leetcode/BinaryTreeGeneral/InvertBinaryTree.kt
package org.khun.leetcode.BinaryTreeGeneral

fun invertTree(root: TreeNode?): TreeNode?{
    if (root == null) return null
    val temp = root.left
    root.left = invertTree(root.right)
    root.right = invertTree(temp)
    return root
}

fun main() {

}